iotile_analytics_core/iotile_analytics/core/channels/cloud_api.py

```diff
@@ -114,14 +154,38 @@ def count_streams(self, slugs):
                 data points in this stream.
         """
 
+        return self._stream_counter(slugs)
+
+    def _count_device_streams(self, slugs):
+        info = self._api.device(self._cloud_id).extra.get()
+
+        stream_table = info.get('stream_counts', {})
+
+        # Note the specific logic here of is not False.  We explictly want to exclude has_streamid is True and is None
+        # since those corresponds with normal streams like app only streams.
+        normal_streams = [slug for slug in slugs if stream_table.get(slug, {}).get('has_streamid') is not False]
+
+        normal_counts = self._count_generic_streams(normal_streams)
+
+        results = {}
+        for slug in slugs:
+            if slug in normal_counts:
+                results[slug] = normal_counts[slug]
+            elif slug in stream_table:
+                results[slug] = {'points': stream_table[slug].get('data_cnt', 0), 'events': 0}
```

Comment:
Should be:
```
'events': stream_table[slug].get('event_cnt', 0)
```

Reply:
Got it, I didn't realize that was a return option.  Fixing.